controllers/app_controller.go

```diff
@@ -138,17 +139,29 @@ func (r *AppReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *AppReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	predicate, err := predicate.LabelSelectorPredicate(r.LabelSelector)
+	labelPredicate, err := predicate.LabelSelectorPredicate(r.LabelSelector)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	namespacedNamePredicate, err := namespacedNamePredicate(key.MCDexDefaultNamespacedName())
 	if err != nil {
 		return microerror.Mask(err)
 	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.App{}).
-		WithEventFilter(predicate).
+		WithEventFilter(predicate.Or(labelPredicate, namespacedNamePredicate)).
```

Comment:
The predicate selects all the Apps matching the dex-app label selector OR the namespaced name of the giantswarm MC dex-app.